home/views.py

```diff
@@ -143,3 +144,30 @@ def post(self, request, *args, **kwargs):
         else:
             messages.error(self.request, gettext("Something went wrong."))
             return self.form_invalid(form)
+
+
+class UserSurveyResponseView(
+    LoginRequiredMixin, UserPassesTestMixin, FormMixin, DetailView
+):
+    model = UserSurveyResponse
+    form_class = UserSurveyResponseForm
+    success_url = reverse_lazy("session_list")
+    template_name = "home/surveys/form.html"
+
+    def test_func(self):
+        survey_response = self.get_object()
+        user = self.request.user
+        return user == survey_response.user
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs["user_survey_response"] = self.get_object()
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        survey = self.get_object().survey
```

Comment:
I don't think it's a terrible idea to select related and prefetch related all the relevant data that the form is going to use.